screens/profilePage.js

```diff
@@ -1,26 +1,199 @@
 
 class profilePage extends Component{
   // TODO: add more user info later once the design has finalized
   constructor(props) {
     super(props);
+    this.state = {
+      userInfo: {},
+      userId: firebase.auth().currentUser.uid,
+      edit: false,
+      new_name: null,
+      new_major: null,
+      new_year: null
+    }
+    this.get_profile_comp = this.get_profile_comp.bind(this);
+    this.edit = this.edit.bind(this)
+    this.save = this.save.bind(this)
+    this.cancel_edit = this.cancel_edit.bind(this)
   }
 
+  edit = () => {
+    this.setState({edit: true});
+  }
+  cancel_edit = () => {
+    this.setState({edit: false});
+  }
+
+  save = () => {
+    var new_userInfo = this.state.userInfo
+    const userId = this.state.userId
+    this.setState({edit: false})
+    const db = firebase.database().ref('/users')
+    if (this.state.new_name !== null) {
+      const new_name = this.state.new_name.split(' ')
+      const first = new_name[0]
+      const last = new_name[1]
+      db.child('/'+userId +'/first_name').set(first)
+      db.child('/'+userId +'/last_name').set(last)
+      new_userInfo.first_name = first
+      new_userInfo.last_name = last
+    }
+    if (this.state.new_major !== new_userInfo.major) {
+      const new_major = this.state.new_major
+      db.child('/'+userId +'/major').set(new_major)
+      new_userInfo.major = new_major
+    }
+    if (this.state.new_year !== new_userInfo.year) {
+      const new_year = this.state.new_year
+      db.child('/'+userId +'/year').set(new_year)
+      new_userInfo.year = new_year
+    }
+    this.setState({userInfo: new_userInfo})
+
+
+  }
+
+
   logout = () => {
     firebase.auth().signOut()
     this.props.navigation.replace('loginPage');
   }
 
+  componentDidMount() {
+    const userId = this.state.userId
+    console.log('userID: ',userId)
+    const db = firebase.database().ref('/users/'+userId);
+    var user = {}
+    db.on('value', (snapshot) => {
+      if (snapshot.exists()) {
+        user = snapshot.toJSON()
+        console.log('user: ',user)
+      }
+      this.setState({userInfo: user,
+                    new_year: user.year,
+                  new_major: user.major,
+                  new_name: user.first_name+' '+user.last_name
+                  })
+      console.log('userInfo: ',this.state.userInfo)
+      });
+  };
+
+  get_profile_comp = () => {
+    const user = firebase.auth().currentUser
+    console.log('user: ',user)
+    if (user.photoURL === '..') {
+      return <Avatar.Image size={65} source={user.profile_picture} style={styles.prof_pic}/>
+    }
+    else {
+      var userInitials = user.displayName.split(' ')
+      userInitials = userInitials[0].substring(0,1) + userInitials[1].substring(0,1)
+      console.log('userInitials: ',userInitials);
+      return <Avatar.Text size={65} label={userInitials} style={styles.prof_pic}/>
```

Comment:
I think there is a profile_picture field that belongs to user object. You can use that to access picture. i.e. https://lh3.googleusercontent.com/a/AATXAJzp5_VNA2AWZyDN5HlRswXVvfrHyo6mN4z_9vRC=s96-c

This may also be helpful because it is up to date with the google profile picture.

Reply:
Can't use user.profile_picture because there was a sync issue with when render called it for the first time. Will look into your idea though

Reply:
hmmm if it is too difficult, then we can save that for later ticket. I like this way as well.